sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -511,16 +658,19 @@ def query_modifier(query: exp.Select) -> exp.Select:
                 return query
             return query.where(self._snapshot_name_version_filter(snapshots))
 
-        return self._get_snapshot_intervals(query_modifier)[1]
+        return self._get_snapshot_intervals(query_modifier=query_modifier)[1]
 
     def _get_snapshot_intervals(
-        self, query_modifier: t.Optional[t.Callable[[exp.Select], exp.Select]]
+        self,
+        snapshot_ids: t.Optional[t.Iterable[SnapshotIdLike]] = None,
+        query_modifier: t.Optional[t.Callable[[exp.Select], exp.Select]] = None,
     ) -> t.Tuple[t.Set[str], t.List[SnapshotIntervals]]:
         query = (
             exp.select(
                 "id", "name", "identifier", "version", "start_ts", "end_ts", "is_dev", "is_removed"
             )
             .from_(self.intervals_table)
+            .where(None if not snapshot_ids else self._snapshot_id_filter(snapshot_ids))
```

Comment:
I think we shouldn't fetch intervals if the `snapshot_ids` list is empty. That might be an error that we'd mask.